File: src/main/java/top/techmczs/cuitxcpctool/entity/Team.java
/*
 * Copyright (C) 2018-2026 Modding Craft ZBD Studio.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package top.techmczs.cuitxcpctool.entity;

import com.alibaba.excel.annotation.ExcelProperty;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;

/**
 * 对应数据库表 dj_team
 */
@Data
@TableName("dj_team") // 绑定表名
public class Team implements Serializable {

    @ExcelProperty("examNumber") // Excel列名
    @TableId
    private String examNumber;

    @ExcelProperty("teamName") // Excel列名
    private String teamName;

    @ExcelProperty("school")
    private String school;

    @ExcelProperty("position")
    private String position;

    @ExcelProperty("account")
    private String account;

    @ExcelProperty("password")
    private String password;

    @ExcelProperty("teammate")
    private String teammate;

    @ExcelProperty("coach")
    private String coach;

}

File: xcpcio/packages/libs/types/src/team.ts
/*
 * Copyright (C) 2018-2026 Modding Craft ZBD Studio.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

import type {Image, Text} from "./basic-types";
import type {Persons} from "./person";

export interface Team {
  id?: string;
  team_id?: string;

  name?: Text;
  team_name?: Text;

  description?: Text;

  organization?: string;
  organization_id?: string;

  group?: Array<string>;
  tag?: Array<string>;

  coach?: Text | Array<Text> | Persons;
  coaches?: Text | Array<Text> | Persons;
  members?: Text | Array<Text> | Persons;

  official?: boolean;
  unofficial?: boolean;
  girl?: boolean;

  badge?: Image;

  missing_photo?: boolean;
  photo?: Image;

  location?: string;
  icpc_id?: string;

  ip?: string;
}

export type Teams = Array<Team> | Record<string /* team_id */, Team>;
